### PR TITLE
Dependencies check

### PR DESCRIPTION
**Pull request by mkunkel**
Adds check for the existence of .ruby-version.example and .ruby-gemset.example
Refactors Options to ensure all checks automatically have a skip option, with short options
Also minor output formatting

### CODE REVIEW

File: exe/options.rb [hunk 1]
```diff
@@ -10,23 +10,34 @@ def self.parse!
...
-      end
+  def self.override_names
+    {
+      CurrentVersion: 'Version'
+    }
+  end
 
-    end.parse!
-    options
+  def self.get_shorty(name, opts)
```
Comment (edubkendo) [inline exe/options.rb]: cute

File: lib/bobo/checks/dependencies.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        missing = files.reject { |file| File.exist?(file) }.join(' & ')
+        puts "Missing files: #{missing}"
+      end
+
+      def check_name
+        'Dependencies'
+      end
+
+      def files
+        %w(.ruby-version.example .ruby-gemset.example)
```
Comment (edubkendo) [inline lib/bobo/checks/dependencies.rb]: do we want to force projects to have gemset examples? it's fine, its just half of us dont use rvm so we'd need help from someone else to generate the example file. if it is on the checklist, maybe this isn't the best place to have the discussion.

File: exe/options.rb [hunk 3]
```diff
@@ -10,23 +10,34 @@ def self.parse!
         options[:verbose] = v
       end
 
-      opts.on("--[no-]skip-server", "Skip server check") do |v|
-        options[:skip_server] = v
-      end
-
-      opts.on("--[no-]skip-version", "Skip version check") do |v|
-        options[:skip_version] = v
+      Bobo::Checks.constants.each do |klass|
```
Comment (edubkendo) [inline exe/options.rb]: looklook
Nice!

**Comment by edubkendo** [comment]
Zero problems. Like how DRY this is. Feel free to merge, just wanted to register my question while I was thinking about it.

kannathumbb

File: lib/bobo/checks/dependencies.rb (hunk 2 above)
Comment (mkunkel) [inline lib/bobo/checks/dependencies.rb]: It is on the checklist. I hadn't noticed that our projects even have the gemset examples. These example files are used directly to deploy projects.

Comment (edubkendo) [inline lib/bobo/checks/dependencies.rb]: Maybe @leejones can clarify for us.

**Pull request merged**